src/SMU-Keithley_6430/main.py

```
@@ -178,8 +237,86 @@ def configure(self) -> None:
+
+            number_of_points = int(self.port.query(f":SOURce:SWEep:POINts?"))
+
+        else:  # self.list_type == "Custom"
+            self.port.write(f":SOURce:{source_type}:MODE LIST")
+            # ensure that custom values are formatted as comma-separated list
+            custom_values_list = [v.strip() for v in self.list_custom_values.split(",")]
+            for value in custom_values_list:
+                try:
+                    float(value)  # validate that each value can be parsed as float
+                except ValueError:
+                    raise ValueError(f"Invalid custom list value: '{value}' is not a number")
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)